build.gradle.kts

```diff
@@ -215,14 +215,18 @@ tasks {
 
 	compileJava {
 		options.encoding = "UTF-8"
-		options.compilerArgs.add("-Werror")
-		options.compilerArgs.add("-Xlint:all")
+		if (!modularBuild.toBoolean()) {
+			options.compilerArgs.add("-Werror")
+			options.compilerArgs.add("-Xlint:all")
+		}
```

Comment:
@Michael1993, instead of completely ignoring warnings, you could probably in general ignore warnings for the `exports` group, see [`javac` documentation](https://docs.oracle.com/en/java/javase/17/docs/specs/man/javac.html#option-Xlint-custom):
```
options.compilerArgs.add("-Xlint:all,-exports")
```

However, except for the `...Extension.class` usage in annotations, many of these warnings seem legit. But it might make sense to address this in a separate pull request.

Reply:
After going over the warnings, I decided, they're not helpful (see 5425bc5), so we ignore them.